classes/util/UtilMethods.php

```diff
@@ -223,23 +223,25 @@ private function getProductDetails( $orderline, $taxConfiguration, $couponDiscou
 		// generate compensation amount orderline using product id and amount
 		$compensationOrderline = $this->compensationAmountOrderline( $productId, $compensationAmount );
 		// generate linedate with all the calculated parameters
-		$linedata[] = array(
-			'description' => $orderline['name'],
-			'itemId'      => $productId,
-			'quantity'    => $productQuantity,
-			'unitPrice'   => number_format( $productPrice, 2, '.', '' ),
-			'taxAmount'   => number_format( $taxAmount * $productQuantity, 2, '.', '' ),
-			'taxPercent'  => number_format( ( $taxAmount / $productPrice ) * 100, 2, '.', '' ),
-			'unitCode'    => $unitCode,
-			'discount'    => number_format( $discountPercentage, 2, '.', '' ),
-			'goodsType'   => 'handling',
-			'imageUrl'    => wp_get_attachment_url( get_post_thumbnail_id( $singleProduct->get_id() ) ),
-			'productUrl'  => get_permalink( $singleProduct->get_id() ),
+		$orderLine = new OrderLine(
+			$orderline['name'],
+			$productId,
+			$productQuantity,
+			number_format( $productPrice, 2, '.', '' )
 		);
 
+		$orderLine->discount   = number_format( $discountPercentage, 2, '.', '' );
+		$orderLine->taxAmount  = number_format( $taxAmount * $productQuantity, 2, '.', '' );
+		$orderLine->taxPercent = number_format( ( $taxAmount / $productPrice ) * 100, 2, '.', '' );
```

Comment:
`number_format` is not the proper way to round floats, please correct this in all instances.